l20-spring/src/main/java/ua/lits/l20spring/controller/UserRestController.java

```diff
@@ -0,0 +1,54 @@
+package ua.lits.l20spring.controller;
+
+import ua.lits.l20spring.dto.UserDTO;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import ua.lits.l20spring.service.UserService;
+
+import javax.validation.Valid;
+import java.util.Collection;
+
+@AllArgsConstructor
+@RestController
+@RequestMapping("/api/users")
+public class UserRestController {
+
+    private UserService userService;
+
+    @GetMapping
+    public Collection<UserDTO> findAll() {
+        return userService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public UserDTO findById(@PathVariable Long id) {
+        return userService.findById(id);
+    }
+
+    @GetMapping("/byEmail/{email}")
+    public UserDTO findByEmail(@PathVariable String email) {
```

Comment:
is better not to contains email on url, use /users/user?email=

Reply:
Yes, but it's impossible to attach a few methods with RequestParam to one url, e.g. I must use unique mapping anyway. That's why I decided to leave it as is